test/recursionGuard.js

```diff
@@ -0,0 +1,261 @@
+const t = require('../test-lib/test.js');
+const assert = require('assert');
+const cuid = require('cuid');
+
+describe('Utils', function() {
```

Comment:
Looks great. One minor change: this file should "describe" recursion guard, not Utils.